lib_eio/process.mli

```diff
@@ -0,0 +1,148 @@
+(** Example:
+    {[
+      # Eio_main.run @@ fun env ->
+        let proc_mgr = Eio.Stdenv.process_mgr env in
+        Eio.Process.parse_out proc_mgr Eio.Buf_read.line ["echo"; "hello"]
+    ]}
+ *)
+
+(** {2 Status and error types} *)
+
+type exit_status = [
+  | `Exited of int      (** Process exited with the given return code. *)
+  | `Signaled of int    (** Process was killed by the given signal. *)
+]
+
+type status = [
+  | exit_status
+  | `Stopped of int     (** Process was stopped (paused) by the given signal. *)
+]
+
+val pp_status : status Fmt.t
+
+type error =
+  | Executable_not_found of string      (** The requested executable does not exist. *)
+  | Child_error of exit_status          (** The child exited with an error status. *)
+
+type Exn.err += E of error
+
+val err : error -> exn
+(** [err e] is [Eio.Exn.create (E e)] *)
+
+val pp_args : string list Fmt.t
+(** Formats a list of arguments, quoting any that might cause confusion to the reader.
+
+    This is intended for use in error messages and logging.*)
+
+(** {2 Processes} *)
+
+(** A child process. *)
+class virtual t : object
+    method virtual pid : int
+    method virtual await : exit_status
+    method virtual signal : int -> unit
+end
+
+val pid : #t -> int
+(** [pid t] is the process ID of [t]. *)
+
+val await : #t -> exit_status
+(** [await t] waits for subprocess [t] to exit and then reports the status. *)
+
+val await_exn : #t -> unit
+(** Like {! await} except an exception is raised if the status is not [`Exited 0]. *)
+
+val signal : #t -> int -> unit
+(** [signal t i] sends the signal [i] to process [t].
+
+    If the process has already exited then this does nothing
+    (it will not signal a different process, even if the PID has been reused).
+
+    See {!Sys} for the signal numbers. *)
+
+(** {2 Spawning processes} *)
+
+class virtual mgr : object
+  method virtual pipe :
+    sw:Switch.t ->
+    <Flow.source; Flow.close> * <Flow.sink; Flow.close>
+
+  method virtual spawn :
+    sw:Switch.t ->
+    ?cwd:Fs.dir Path.t ->
+    ?stdin:Flow.source ->
+    ?stdout:Flow.sink ->
+    ?stderr:Flow.sink ->
+    ?env:string array ->
+    ?executable:string ->
+    string list ->
+    t
+end
+(** A process manager capable of spawning new processes. *)
```

Comment:
I think (and this is on me from the initial commits) we've used 

 - Process
 - Subprocess
 - Sub-process
 - Child process

in different places to refer to the same thing. Sometimes it is okay, e.g. to describe the _process ID_. But here for example I'm guessing we should say a "manager capable of spawning new **subprocesses**" given that for the function `spawn` we say "creates a new subprocess". Perhaps this doesn't matter too much and the intent is clear?